docs/src/model_creation/dsl_advanced.md

```diff
@@ -272,6 +272,36 @@ sol = solve(oprob)
 plot(sol)
 ```
 
+### [Turning off species, parameter, and variable inferring](@id dsl_advanced_options_require_dec)
+In some cases it may be desirable for Catalyst to not infer species and parameters from the DSL, as in the case of reaction networks with very many variables, or as a sanity check that variable names are written correctly. To turn off inferring, simply add the `@require_declaration` macro to one of the lines of the `@reaction_network` declaration. Having this macro means that every single variable, species, or parameter will have to be explicitly declared using the `@variable`, `@species`, or `@parameter` macro. In the case that the DSL parser encounters an undeclared symbolic in this case, it will error with an `UndeclaredSymbolicError` and print the reaction or equation that the undeclared symbolic was found in. 
+
+```@example dsl_advanced_no_infer
+using Catalyst
+# The following case will throw an UndeclaredSymbolicError.
+rn = @reaction_network begin
+    @require_declaration
+    (k1, k2), A <--> B
+end
```

Comment:
To keep Documenter.jl from failing the build you probably need to catch the error here. (Otherwise just don't make this dynamic and manually paste in the output.)